samcli/lib/cookiecutter/exceptions.py

```diff
@@ -0,0 +1,18 @@
+""" Exceptions raised from the cookiecutter workflow"""
+
+
+class CookiecutterErrorException(Exception):
```

Comment:
I wonder should we make this class private (by prefixing it with `_`) so people won't mistakenly use it.

Reply:
but people may need to use it to capture `GenerateProjectFailedError` and `InvalidLocationError` and take an action.

Reply:
Oh I see. There seems to have no other way